sdl_viewer/src/lib.rs

```diff
@@ -307,249 +298,226 @@ fn load_camera(index: usize, pose_path: &Option<PathBuf>, camera: &mut Camera) {
     camera.set_state(states.states[index]);
 }
 
-impl SdlViewer {
-    pub fn new() -> Self {
-        SdlViewer {
-            octree_factories: FnvHashMap::default(),
-        }
+pub fn run(octree_factory: OctreeFactory) {
```

Comment:
It may be cleaner (interface wise) to pass a &str to the run method and internally hold the octree factory. What do you think?

Reply:
I figured it can only make sense when the fork in the private repo comes back to OS, because then the main customizes the OctreeFactory and the rest of the sdl functionality remains the same. 
I also tried to put the OctreeFactory as a part of the (once implemented, now disappeared) SdlViewer struct but I had compiler problems.
The problems:
-by register_octree_factory: hashtable and copy trait don't go together.
Do you have some suggestion in the implementation of register_octree_factory? If not and if you also agree I'd pospone this when the refactoring of sdl_viewer takes place, cause there is a lot going on in the lib.rs file.

Reply:
Ok, for me it's fine to postpone it.

Reply:
I'm adding a todo to take care of that in the future